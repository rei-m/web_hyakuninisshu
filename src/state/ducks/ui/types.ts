import { Action } from 'redux';
import { Color, Kimariji } from '@src/types';
import * as constants from './constants';

export type OPEN_KARUTAS_FILTER_TYPE = typeof constants.OPEN_KARUTAS_FILTER_NAME;

export type CLOSE_KARUTAS_FILTER_TYPE = typeof constants.CLOSE_KARUTAS_FILTER_NAME;

export type TOGGLE_KARUTAS_COLOR_TYPE = typeof constants.TOGGLE_KARUTAS_COLOR_NAME;

export type TOGGLE_KARUTAS_KIMARIJI_TYPE = typeof constants.TOGGLE_KARUTAS_KIMARIJI_NAME;

export interface OpenKarutasFilterAction extends Action {
  readonly type: OPEN_KARUTAS_FILTER_TYPE;
}

export interface CloseKarutasFilterAction extends Action {
  readonly type: CLOSE_KARUTAS_FILTER_TYPE;
}

export interface OKarutasFilterAction extends Action {
  readonly type: OPEN_KARUTAS_FILTER_TYPE;
}

export interface ToggleKarutasColorAction extends Action {
  readonly type: TOGGLE_KARUTAS_COLOR_TYPE;
  readonly payload: {
    readonly color: Color;
    readonly checked: boolean;
  };
}

export interface ToggleKarutasKimarijiAction extends Action {
  readonly type: TOGGLE_KARUTAS_KIMARIJI_TYPE;
  readonly payload: {
    readonly kimariji: Kimariji;
    readonly checked: boolean;
  };
}

export type Actions =
  | OpenKarutasFilterAction
  | CloseKarutasFilterAction
  | ToggleKarutasColorAction
  | ToggleKarutasKimarijiAction;

export interface KarutasFilter {
  readonly colors: Array<{ color: Color; checked: boolean }>;
  readonly kimarijis: Array<{ kimariji: Kimariji; checked: boolean }>;
}

export interface UiState {
  readonly karutasFilter: KarutasFilter & { readonly open: boolean };
}
