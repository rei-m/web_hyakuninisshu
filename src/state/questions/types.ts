import { Action } from 'redux';
import { Kimariji, Color, QuestionId, KarutaNo, Karuta } from '@src/domain/models';
import * as constants from './constants';
import { Meta, Payload } from '..';

type Condition<T> = {
  values: Array<T>;
};

export type RangeFromCondition = 1 | 11 | 21 | 31 | 41 | 51 | 61 | 71 | 81 | 91;
export const RangeFromCondition: Condition<RangeFromCondition> = {
  values: [1, 11, 21, 31, 41, 51, 61, 71, 81, 91],
};

export type RangeToCondition = 10 | 20 | 30 | 40 | 50 | 60 | 70 | 80 | 90 | 100;
export const RangeToCondition: Condition<RangeToCondition> = {
  values: [10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
};

export type KimarijiCondition = null | Kimariji;
export const KimarijiCondition: Condition<KimarijiCondition> = {
  values: [null, 1, 2, 3, 4, 5, 6],
};

export type ColorCondition = null | Color;
export const ColorCondition: Condition<ColorCondition> = {
  values: [null, 'blue', 'pink', 'yellow', 'green', 'orange'],
};

export type KarutaStyleCondition = 'kanji' | 'kana';
export const KarutaStyleCondition: Condition<KarutaStyleCondition> = {
  values: ['kanji', 'kana'],
};

export type QuestionAnimCondition = 'none' | 'slow' | 'normal' | 'fast';
export const QuestionAnimCondition: Condition<QuestionAnimCondition> & {
  toDulation: (value: QuestionAnimCondition) => number;
} = {
  values: ['none', 'slow', 'normal', 'fast'],
  toDulation: (value: QuestionAnimCondition) => {
    switch (value) {
      case 'normal':
        return 0.6;
      case 'fast':
        return 0.3;
      case 'slow':
        return 1.0;
      default:
        return 0;
    }
  },
};

export type YomiFuda = Readonly<{
  karutaNo: KarutaNo;
  shoku: string;
  niku: string;
  sanku: string;
}>;

export type ToriFuda = Readonly<{
  karutaNo: KarutaNo;
  shiku: string;
  kekku: string;
}>;

export type StartTrainingAction = Action<typeof constants.START_TRAINING_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
    totalCount: number;
  }> &
  Meta<{
    rangeFrom: RangeFromCondition;
    rangeTo: RangeToCondition;
    kimariji: KimarijiCondition;
    color: ColorCondition;
    kamiNoKuStyle: KarutaStyleCondition;
    shimoNoKuStyle: KarutaStyleCondition;
    questionAnim: QuestionAnimCondition;
  }>;

export type RestartTrainingAction = Action<typeof constants.RESTART_TRAINING_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
    totalCount: number;
  }>;

export type StartExamAction = Action<typeof constants.START_EXAM_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
    totalCount: number;
  }>;

export type StartQuestionAction = Action<typeof constants.START_QUESTION_NAME> &
  Payload<{
    questionId: QuestionId;
    content: {
      yomiFuda: YomiFuda;
      toriFudaList: [ToriFuda, ToriFuda, ToriFuda, ToriFuda];
    };
  }>;

export type ResetQuestionAction = Action<typeof constants.RESET_QUESTION_NAME>;

export type FinishQuestionAction = Action<typeof constants.FINISH_QUESTION_NAME> &
  Payload<{
    correctCount: number;
    averageAnswerSecond: number;
    answerList: Array<{
      questionId: QuestionId;
      isCorrect: boolean;
      correctKaruta: Karuta;
    }>;
  }>;

export type AnswerQuestionAction = Action<typeof constants.ANSWER_QUESTION_NAME> &
  Payload<{
    isCorrect: boolean;
    selectedKarutaNo: KarutaNo;
    correctKaruta: Karuta;
  }>;

export type ConfirmCorrectAction = Action<typeof constants.CONFIRM_CORRECT_NAME>;

export type OpenNextQuestionAction = Action<typeof constants.OPEN_NEXT_QUESTION_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
  }>;

export type Actions =
  | StartTrainingAction
  | RestartTrainingAction
  | StartExamAction
  | StartQuestionAction
  | ResetQuestionAction
  | FinishQuestionAction
  | AnswerQuestionAction
  | ConfirmCorrectAction
  | OpenNextQuestionAction;

export type ActionCreator = {
  /**
   * 指定した条件から問題を作成/保存してStartTrainingActionを返す
   *
   * @param rangeFrom 出題範囲From
   * @param rangeTo 出題範囲To
   * @param kimariji 決まり字条件
   * @param color 色条件
   * @param kamiNoKuStyle 上の句表示形式
   * @param shimoNoKuStyle 下の句表示形式
   * @param questionAnim 問題の表示速度
   *
   * @returns {StartTrainingAction}
   * @throws {Error} 指定した条件に当てはまる歌が無い場合
   */
  startTraining: (
    rangeFrom: RangeFromCondition,
    rangeTo: RangeToCondition,
    kimariji: KimarijiCondition,
    color: ColorCondition,
    kamiNoKuStyle: KarutaStyleCondition,
    shimoNoKuStyle: KarutaStyleCondition,
    questionAnim: QuestionAnimCondition
  ) => StartTrainingAction;

  /**
   * 回答済の問題の中から間違えた問題をもとに問題を作成/保存してRestartTrainingActionを返す
   *
   * @returns {RestartTrainingAction}
   * @throws {Error} 指定した条件に当てはまる歌が無い場合
   */
  restartTraining: () => RestartTrainingAction;

  /**
   * 全ての歌を対象に問題を作成/保存してStartExamActionを返す
   *
   * @returns {StartExamAction}
   * @throws {Error} 指定した条件に当てはまる歌が無い場合
   */
  startExam: () => StartExamAction;

  /**
   * 問題の回答を開始してStartQuestionActionを返す
   *
   * @returns {StartQuestionAction}
   * @throws {Error} 指定した問題がない場合
   */
  startQuestion: (
    currentQuestionId: QuestionId,
    kamiNoKuStyle: KarutaStyleCondition,
    shimoNoKuStyle: KarutaStyleCondition,
    startDate: Date
  ) => StartQuestionAction;

  /**
   * 問題の回答を判定してAnswerQuestionActionを返す
   *
   * @returns {AnswerQuestionAction}
   * @throws {Error} 指定した問題がない場合
   * @throws {Error} 問題の回答が開始されていない場合
   */
  answerQuestion: (currentQuestionId: QuestionId, toriFuda: ToriFuda, answerDate: Date) => AnswerQuestionAction;

  /**
   * ConfirmCorrectActionを返す
   *
   * @returns {ConfirmCorrectAction}
   */
  confirmCorrect: () => ConfirmCorrectAction;

  /**
   * 次の問題を取得してOpenNextQuestionActionを返す
   *
   * @returns {OpenNextQuestionAction}
   * @throws {Error} 次の問題がない場合
   */
  openNextQuestionAction: (currentQuestionId: QuestionId) => OpenNextQuestionAction;

  /**
   * 問題を初期状態に戻してResetQuestionActionを返す
   *
   * @returns {ResetQuestionAction}
   */
  resetQuestion: () => ResetQuestionAction;

  /**
   * 問題を終了して結果を見る
   *
   * @returns {FinishQuestionAction}
   */
  finishQuestion: () => FinishQuestionAction;
};

export type State = Readonly<{
  state: 'waiting' | 'ready' | 'playing' | 'confirm' | 'finished';
  totalCount: number;
  currentPosition: number;
  currentQuestion?: {
    questionId: QuestionId;
    content?: {
      yomiFuda: YomiFuda;
      toriFudaList: [ToriFuda, ToriFuda, ToriFuda, ToriFuda];
    };
    answer?: {
      isCorrect: boolean;
      selectedKarutaNo: KarutaNo;
      correctKaruta: Karuta;
    };
  };
  result?: {
    correctCount: number;
    averageAnswerSecond: number;
    answerList: Array<{
      questionId: QuestionId;
      isCorrect: boolean;
      correctKaruta: Karuta;
    }>;
  };
  trainingCondition: {
    rangeFrom: RangeFromCondition;
    rangeTo: RangeToCondition;
    kimariji: KimarijiCondition;
    color: ColorCondition;
    kamiNoKuStyle: KarutaStyleCondition;
    shimoNoKuStyle: KarutaStyleCondition;
    questionAnim: QuestionAnimCondition;
  };
}>;
