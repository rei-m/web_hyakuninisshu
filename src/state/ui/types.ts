import { Action } from 'redux';
import * as constants from './constants';
import { Color, Kimariji } from '@src/domain/models';
import { Payload } from '..';

export type KarutasFilter = Readonly<{
  colors: Array<{ color: Color; checked: boolean }>;
  kimarijis: Array<{ kimariji: Kimariji; checked: boolean }>;
}>;

export type OpenKarutasFilterAction = Action<typeof constants.OPEN_KARUTAS_FILTER_NAME>;

export type CloseKarutasFilterAction = Action<typeof constants.CLOSE_KARUTAS_FILTER_NAME>;

export type ToggleKarutasColorAction = Action<typeof constants.TOGGLE_KARUTAS_COLOR_NAME> &
  Payload<{
    color: Color;
    checked: boolean;
  }>;

export type ToggleKarutasKimarijiAction = Action<typeof constants.TOGGLE_KARUTAS_KIMARIJI_NAME> &
  Payload<{
    kimariji: Kimariji;
    checked: boolean;
  }>;

export type Actions =
  | OpenKarutasFilterAction
  | CloseKarutasFilterAction
  | ToggleKarutasColorAction
  | ToggleKarutasKimarijiAction;

export type ActionCreator = {
  /**
   * OpenKarutasFilterActionを返す
   *
   * @returns {OpenKarutasFilterAction}
   */
  openKarutasFilter: () => OpenKarutasFilterAction;

  /**
   * CloseKarutasFilterActionを返す
   *
   * @returns {CloseKarutasFilterAction}
   */
  closeKarutasFilter: () => CloseKarutasFilterAction;

  /**
   * ToggleKarutasColorActionを返す
   *
   * @param color 歌の色
   * @param checked 表示する/しない
   *
   * @returns {ToggleKarutasColorAction}
   */
  toggleKarutasColor: (color: Color, checked: boolean) => ToggleKarutasColorAction;

  /**
   * ToggleKarutasKimarijiActionを返す
   *
   * @param kimariji 歌の決まり字
   * @param checked 表示する/しない
   *
   * @returns {ToggleKarutasKimarijiAction}
   */
  toggleKarutasKimariji: (kimariji: Kimariji, checked: boolean) => ToggleKarutasKimarijiAction;
};

export type State = Readonly<{
  karutasFilter: KarutasFilter & { open: boolean };
}>;
