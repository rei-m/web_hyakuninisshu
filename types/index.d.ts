declare module '*.png'
declare module '*.jpg'

// http://ideasintosoftware.com/typescript-advanced-tricks/s
type Diff<T extends string, U extends string> = ({[P in T]: P } & {[P in U]: never } & { [x: string]: never })[T];  
type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>; 
