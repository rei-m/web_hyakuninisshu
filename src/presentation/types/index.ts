import { RouteComponentProps } from '@reach/router';

export type SiteMetaData = {
  title: string;
  description: string;
  author: string;
};

export type GeneratedPageComponentProps<T = {}> = { pageContext: T } & RouteComponentProps<T>;

export type AppErrorType = 'Network' | 'NotFound' | 'Unknown';

export class AppError implements Error {
  private _message: string;
  private _type: AppErrorType;

  constructor(message: string, type: AppErrorType) {
    this._message = message;
    this._type = type;
  }

  public get name() {
    return 'AppError';
  }

  public get message() {
    return this._message;
  }

  public get type() {
    return this._type;
  }

  public toString() {
    return `${this.name}: ${this.message}`;
  }
}
