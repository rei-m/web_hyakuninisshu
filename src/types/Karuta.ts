interface Karuta {
  id: number;
  imageNo: string;
}

export default Karuta;
