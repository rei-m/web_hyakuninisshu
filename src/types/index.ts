export interface Karuta {
  id: number;
  imageNo: string;
}

export interface Question {
  correctKaruta: Karuta;
  choices: Karuta[];
}

export interface OptionItem<T> {
  readonly name: string;
  readonly value: T;
}
