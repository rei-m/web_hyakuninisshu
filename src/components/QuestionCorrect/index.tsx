import * as React from 'react';
import styled from 'styled-components';
import { withState } from 'recompose';
import { Dialog } from '@blueprintjs/core';
import { withAppTheme } from '@src/styles';
import { Karuta } from '@src/types';
import Tatami from '@src/components/Tatami';
import KarutaCard from '@src/components/KarutaCard';
import { toKarutaIdString, toKimarijiString } from '@src/components/helper';

export interface QuestionCorrectProps {
  readonly karuta: Karuta;
  readonly isAllAnswered: boolean;
  readonly onClickGoToNext: () => void;
  readonly onClickGoToResult: () => void;
}

const Frame = Tatami.extend`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: calc(100vh - ${({ theme }) => theme.headerHeight});

  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    min-height: calc(100vh - ${({ theme }) => theme.headerHeightWide});
  }
`;

const Header = styled.div`
  position: relative;
  width: 160px;
`;

const NumberAndKimariji = withAppTheme(styled.div)`
  padding: ${({ theme }) => `${theme.spacing1x} ${theme.spacing2x}`};
  border: 1px solid #d3d3d3;
  width: 160px;
  margin: 0 auto;
  font-size: 1.2rem;
  background-color: #ffffff;
`;

const OpenDetail = styled.button`
  position: absolute;
  top: 0;
  right: -48px;
  height: 33px;
  width: 33px;
`;

const KarutaFrame = withAppTheme(styled.div)`
  width: 200px;
  height: 260px;
  border: 6px solid ${({ theme }) => theme.colorPrimaryDark};
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: ${({ theme }) => theme.spacing2x} auto 0 auto;
  background-color: ${({ theme }) => theme.colorThin};
  box-shadow: 0 -1px 5px rgba(0, 0, 0, 0.26);
  font-family: 'Sawarabi Mincho';
`;

const Inner = styled.div`
  display: flex;
  flex-direction: row-reverse;
  justify-content: center;
`;

const Phrase = styled.div`
  width: 1.9rem;
  font-size: 1.9rem;
  line-height: 2rem;
`;

const SecondPhrase = Phrase.extend`
  padding-top: 24px;
  margin-right: 8px;
`;

const ThirdPhrase = Phrase.extend`
  padding-top: 48px;
  margin-right: 8px;
`;

const FourthPhrase = Phrase.extend`
  padding-top: 32px;
  margin-right: 8px;
`;

const FifthPhrase = Phrase.extend`
  padding-top: 56px;
  margin-right: 8px;
`;

const Creator = styled.div`
  width: 1.6rem;
  margin-right: 16px;
  font-size: 1.4rem;
  line-height: 1.6rem;
  align-self: flex-end;
`;

const NextButton = styled.button`
  margin-top: 32px;
`;

const enhance = withState<{}, boolean, 'opened', 'setOpened'>(
  'opened',
  'setOpened',
  false
);

const QuestionCorrect = enhance<
  QuestionCorrectProps & {
    opened: boolean;
    setOpened: (state: boolean) => boolean;
  }
>(
  ({
    karuta,
    isAllAnswered,
    onClickGoToNext,
    onClickGoToResult,
    opened,
    setOpened
  }) => {
    const onClickOpenDetail = () => {
      setOpened(true);
    };

    const onCloseDialog = () => {
      setOpened(false);
    };

    return (
      <Frame>
        <Header>
          <NumberAndKimariji>
            {toKarutaIdString(karuta.id)} / {toKimarijiString(karuta.kimariji)}
          </NumberAndKimariji>
          <OpenDetail
            onClick={onClickOpenDetail}
            className="pt-button"
            data-test="open-detail"
          >
            詳
          </OpenDetail>
        </Header>
        <KarutaFrame>
          <Inner>
            <Phrase>{karuta.firstKanji}</Phrase>
            <SecondPhrase>{karuta.secondKanji}</SecondPhrase>
            <ThirdPhrase>{karuta.thirdKanji}</ThirdPhrase>
            <FourthPhrase>{karuta.fourthKanji}</FourthPhrase>
            <FifthPhrase>{karuta.fifthKanji}</FifthPhrase>
            <Creator>{karuta.creator}</Creator>
          </Inner>
        </KarutaFrame>
        {isAllAnswered ? (
          <NextButton
            onClick={onClickGoToResult}
            className="pt-button pt-large pt-icon-double-chevron-right"
            data-test="go-to-result"
          >
            結果を見る
          </NextButton>
        ) : (
          <NextButton
            onClick={onClickGoToNext}
            className="pt-button pt-large pt-icon-double-chevron-right"
            data-test="go-to-next"
          >
            次へ進む
          </NextButton>
        )}
        <Dialog
          isOpen={opened}
          onClose={onCloseDialog}
          title="正解"
          style={{
            maxWidth: 380,
            padding: 0,
            width: '80vw'
          }}
        >
          <KarutaCard karuta={karuta} />
        </Dialog>
      </Frame>
    );
  }
);

export default QuestionCorrect;
