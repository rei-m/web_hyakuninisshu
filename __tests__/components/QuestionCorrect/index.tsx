import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { shallow } from 'enzyme';
import { create } from '@test/factories';
import { sel, setUpQueryOnce } from '@test/helpers';
import QuestionCorrect, { Props, QueryData as QuestionCorrectQueryData } from '@src/components/QuestionCorrect';
import { Karuta } from '@src/types';

describe('<QuestionCorrect />', () => {
  let baseProps: Props;

  beforeEach(() => {
    setUpQueryOnce<QuestionCorrectQueryData>({
      questionCorrectBGImage: {
        publicURL: '/tatami.png',
      },
    });

    baseProps = {
      isAllAnswered: false,
      karuta: create<Karuta>('karuta', {
        no: 1,
      }),
      onClickGoToNext: jest.fn(),
      onClickGoToResult: jest.fn(),
    };
  });

  it('should render component when not finished', () => {
    const renderer = ReactTestRenderer.create(<QuestionCorrect {...baseProps} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when finished', () => {
    const props = { ...baseProps, isAllAnswered: true };
    const renderer = ReactTestRenderer.create(<QuestionCorrect {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should fire onClickGoToNext when next clicked', () => {
    const wrapper = shallow(<QuestionCorrect {...baseProps} />)
      .dive()
      .dive();
    wrapper.find(sel('go-to-next')).simulate('click');
    expect(baseProps.onClickGoToNext).toHaveBeenCalled();
  });

  it('should fire onClickGoToResult when next clicked', () => {
    const props = { ...baseProps, isAllAnswered: true };
    const wrapper = shallow(<QuestionCorrect {...props} />)
      .dive()
      .dive();
    wrapper.find(sel('go-to-result')).simulate('click');
    expect(baseProps.onClickGoToResult).toHaveBeenCalled();
  });

  it('should set state when OpenDetail clicked', () => {
    const wrapper = shallow(<QuestionCorrect {...baseProps} />);
    expect(wrapper.state()).toEqual({ stateValue: false });
    wrapper
      .dive()
      .dive()
      .find(sel('open-detail'))
      .simulate('click');
    expect(wrapper.state()).toEqual({ stateValue: true });
  });
});
