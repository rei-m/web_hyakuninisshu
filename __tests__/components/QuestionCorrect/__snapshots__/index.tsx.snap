// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionCorrect /> should render component when finished 1`] = `
<div
  className="sc-gzVnrw hwtXpF"
>
  <div
    className="sc-htoDjs lepjQV"
  >
    <div
      className="sc-dnqmqq jpBwGm"
    >
      一番
       / 
      一字決まり
    </div>
    <button
      className="bp3-button sc-iwsKbI fSyGkH"
      data-test="open-detail"
      onClick={[Function]}
    >
      詳
    </button>
  </div>
  <div
    className="sc-gZMcBi hGOwDf"
  >
    <div
      className="sc-gqjmRU jKiVnA"
    >
      <div
        className="sc-VigVT dNBSaP"
      >
        初句
      </div>
      <div
        className="sc-jTzLTM jDOuIs"
      >
        二句
      </div>
      <div
        className="sc-fjdhpX fwTIxm"
      >
        三句
      </div>
      <div
        className="sc-jzJRlG YovCa"
      >
        四句
      </div>
      <div
        className="sc-cSHVUG iYZJMh"
      >
        五句
      </div>
      <div
        className="sc-kAzzGY ewOkip"
      >
        詠み手
      </div>
    </div>
  </div>
  <button
    className="bp3-button bp3-large bp3-icon-double-chevron-right sc-chPdSV hiOXiY"
    data-test="go-to-result"
    onClick={[MockFunction]}
  >
    結果を見る
  </button>
</div>
`;

exports[`<QuestionCorrect /> should render component when not finished 1`] = `
<div
  className="sc-gzVnrw hwtXpF"
>
  <div
    className="sc-htoDjs lepjQV"
  >
    <div
      className="sc-dnqmqq jpBwGm"
    >
      一番
       / 
      一字決まり
    </div>
    <button
      className="bp3-button sc-iwsKbI fSyGkH"
      data-test="open-detail"
      onClick={[Function]}
    >
      詳
    </button>
  </div>
  <div
    className="sc-gZMcBi hGOwDf"
  >
    <div
      className="sc-gqjmRU jKiVnA"
    >
      <div
        className="sc-VigVT dNBSaP"
      >
        初句
      </div>
      <div
        className="sc-jTzLTM jDOuIs"
      >
        二句
      </div>
      <div
        className="sc-fjdhpX fwTIxm"
      >
        三句
      </div>
      <div
        className="sc-jzJRlG YovCa"
      >
        四句
      </div>
      <div
        className="sc-cSHVUG iYZJMh"
      >
        五句
      </div>
      <div
        className="sc-kAzzGY ewOkip"
      >
        詠み手
      </div>
    </div>
  </div>
  <button
    className="bp3-button bp3-large bp3-icon-double-chevron-right sc-chPdSV hiOXiY"
    data-test="go-to-next"
    onClick={[MockFunction]}
  >
    次へ進む
  </button>
</div>
`;
